### PR TITLE
Sp simplify inhibition

### PR DESCRIPTION
**Pull request by breznak**
Fixes: #2627 

Small changes to inhibition I came across during other SP work

### CODE REVIEW

File: tests/unit/nupic/research/spatial_pooler_unit_test.py [hunk 1]
```diff
@@ -1315,7 +1315,7 @@ def testInhibitColumnsLocal(self):
                         #   L  W  W  L  L  W  W   L   L    L
     trueActive = [1, 2, 5, 6]
     active = list(sp._inhibitColumnsLocal(overlaps, density))
-    # self.assertListEqual(trueActive, active)
+    # self.assertListEqual(trueActive, active) #FIXME why was this commented out? should still be used?
```
Comment (breznak) [inline tests/unit/nupic/research/spatial_pooler_unit_test.py]: Someone? It does not work right off, and I don't intend to fix it in this PR, but someone cloud know or we should make an issue for it, or remove the part of the test.

**Comment by breznak** [comment]
@chetan51 I think you were the last to make bigger changes to inhibition, so I'd like your review please.

File: src/nupic/research/spatial_pooler.py [hunk 2]
```diff
@@ -1423,16 +1421,15 @@ def _inhibitColumnsGlobal(self, overlaps, density):
                     of synapses in a "connected state" (connected synapses)
                     that are connected to input bits which are turned on.
     @param density: The fraction of columns to survive inhibition.
+    @return winning columns indices (as a sparse list)
```
Comment (chetan51) [inline src/nupic/research/spatial_pooler.py]: No need to say `(as a sparse list)`.

File: src/nupic/research/spatial_pooler.py [hunk 3]
```diff
@@ -1453,20 +1450,20 @@ def _inhibitColumnsLocal(self, overlaps, density):
                     value is only an intended target. Since the surviving
                     columns are picked in a local fashion, the exact fraction
                     of surviving columns is likely to vary.
+    @return indices of winning columns (as a sparse list)
```
Comment (chetan51) [inline src/nupic/research/spatial_pooler.py]: No need to say `(as a sparse list)`.

File: tests/unit/nupic/research/spatial_pooler_unit_test.py [hunk 4]
```diff
@@ -212,7 +212,7 @@ def testExactOutput(self):
 
     # Get only the active column indices
     spOutput = [i for i, v in enumerate(activeArray) if v != 0]
-    self.assertEqual(spOutput, expectedOutput)
+    self.assertEqual(sorted(spOutput), expectedOutput)
```
Comment (chetan51) [inline tests/unit/nupic/research/spatial_pooler_unit_test.py]: Can also cast them as `set`s and compare them.

File: tests/unit/nupic/research/spatial_pooler_unit_test.py (hunk 1 above)
Comment (chetan51) [inline tests/unit/nupic/research/spatial_pooler_unit_test.py]: We should make an issue for it and mention the issue in a comment here.

**Comment by chetan51** [comment]
Done with review.

File: tests/unit/nupic/research/spatial_pooler_unit_test.py (hunk 1 above)
Comment (breznak) [inline tests/unit/nupic/research/spatial_pooler_unit_test.py]: ok, there's https://github.com/numenta/nupic/issues/2639

**Comment by breznak** [comment]
Thanks @chetan51 , fixed according to your review.

**Comment by chetan51** [comment]
:+1:

**Pull request merged**